design/database.md

```diff
@@ -0,0 +1,48 @@
+# Database Design
+
+## Table: pipeline\_groups
+| Field name | Data type | Extra
+| --- | --- | ---
+| id | integer | `primary_key`
+| name | string | `NOT NULL`
+| inserted\_at | datetime |
+| updated\_at | datetime |
+
+## Table: pipelines
+| Field name | Data type | Extra
+| --- | --- | ---
+| id | integer | `primary_key`
+| name | string | `NOT NULL`
+| pipeline\_group\_id | belongs\_to | `NOT NULL`
+| inserted\_at | datetime |
+| updated\_at | datetime |
+
+## Table: pipelines\_materials
+| Field name | Data type | Extra
+| --- | --- | ---
+| pipeline\_id | integer | `compound_key`
+| material\_id | integer | `compound_key`
+
+## Table: materials
+| Field name | Data type | Extra
+| --- | --- | ---
+| id | integer | `primary_key`
+| type | string | `enum: {git|file|...}`
+| url | string | `{git://...|file://source_pipeline/release.tar.gz}`
```

Comment:
I would not hardcode the specific implementation of the material. (Like if we want pipelines as material or a specific branch.)

Can we turn this into a JSONB structure. Give it an appropriate name (like a metadata field) and then leave it up to the `type` to define the that that would be needed.

Reply:
That sounds like a better way to do it :+1: